__tests__/modal-container.test.tsx

```diff
@@ -17,16 +15,82 @@ describe('ModalContainer component behaviour', () => {
             onCancel: noop,
         };
 
-        return shallow(<ModalContainer {...props} />);
-    }
+        expect(shallow(<ModalContainer {...props} />).length).toEqual(1);
+    });
+
+    test('Clicking cancel calls onCancel', () => {
+
+        const props = {
+            content: str(),
+            onCancel: jest.fn(),
+        };
+
+        const wrapper = shallow(<ModalContainer {...props} />);
+        const button = wrapper.find('.btn-default');
 
-    afterEach(() => {
-        componentWrapper.unmount();
+        expect(button.text()).toEqual('Cancel');
+        
+        button.simulate('click');
+        expect(props.onCancel).toHaveBeenCalled();
     });
 
-    test('Component renders without crashing', () => {
-        componentWrapper = manyWhoMount();
-        expect(componentWrapper.length).toEqual(1);
+    test('Clicking confirm calls onConfirm', () => {
+
+        const props = {
+            content: str(),
+            onConfirm: jest.fn(),
+        };
+
+        const wrapper = shallow(<ModalContainer {...props} />);
+        const button = wrapper.find('.btn-primary');
+
+        expect(button.text()).toEqual('OK');
+        
+        button.simulate('click');
+        expect(props.onConfirm).toHaveBeenCalled();
+    });
+
+    test('Supplied labels are rendered in buttons', () => {
+
+        const props = {
+            content: str(),
+            onConfirm: jest.fn(),
+            onCancel: jest.fn(),
+            confirmLabel: 'aaa',
+            cancelLabel: 'bbb',
+        };
+
+        const wrapper = shallow(<ModalContainer {...props} />);
+        expect(wrapper.find('.btn-primary').text()).toEqual('aaa');
+        expect(wrapper.find('.btn-default').text()).toEqual('bbb');
+    });
+
+    test('Global modal property is set to null when escape is pressed', () => {
+
+        window.manywho.model.setModal = jest.fn();
+
+        const props = {
+            flowKey: 'zzz',
+            content: str(),
+            onConfirm: jest.fn(),
+            onCancel: jest.fn(),
+        };
+
+        const wrapper = shallow(<ModalContainer {...props} />);
+        wrapper.simulate('keyup', { 
+            keyCode: 27,
+        })
+        expect(window.manywho.model.setModal).toHaveBeenCalledWith('zzz', null);
+    });
+
+    test('Supplied content is rendered', () => {
+
+        const props = {
+            content: <input className="my-input" />,
+        };
+
+        const wrapper = shallow(<ModalContainer {...props} />);
+        expect(wrapper.find('.my-input').length).toBe(1);
```

Comment:
Could add some tests to check that header and footer only render when the correct props are supplied.
Otherwise I think you have all the other possibilities.

Reply:
I considered those, but didn't do the header because it relies on manywho.utils.isNullOrEmpty which I would have to mock and so would be controlling the outcome in my mock implementation.

I will add tests for the footer and also whether each button in the footer should be present.

Reply:
Added some more conditional tests